src/ProductConstructionService/ProductConstructionService.WorkItems/WorkItemScopeManager.cs

```diff
@@ -41,7 +45,11 @@ public async Task<WorkItemScope> BeginWorkItemScopeWhenReadyAsync()
 
     private async Task WorkItemFinishedAsync()
     {
-        await _state.SetStoppedIfStoppingAsync();
+        Interlocked.Decrement(ref _activeWorkItems);
+        if (_activeWorkItems == 0)
```

Comment:
Doesn't `Interlocked.Decrement` return the value?

Reply:
Yeah, both `Decrement` and `Increment` return the new value of the variable.

Oh, meaning it can be used in the conditional? I just felt this way was a bit more verbose

Reply:
A bit more verbose but also not atomic anymore? :)

Reply:
That is true, thanks for pointing it out. Will change